### PR TITLE
Added some STL headers to stl.modulemap

### PR DESCRIPTION
**Pull request by arpi-r**
Added some stl headers to stl.modulemap

### CODE REVIEW

**Comment by arpi-r** [comment]
Hey, I have removed the full path to STL headers. 
Could you please review it?

File: build/unix/stl.modulemap [hunk 1]
```diff
@@ -357,6 +357,56 @@ module "stl" [system] {
...
+  }
+  module "optional" {
+    export *
+    header "optional"
+  }
+  module "shared_mutex" {
+    export *
+    header "shared_mutex"
+  }
+  module "stdlib.h" {
```
Comment (Axel-Naumann) [inline build/unix/stl.modulemap]: This is already (and correctly so) part of the `libc.modulemap`. Please remove here.

File: build/unix/stl.modulemap [hunk 2]
```diff
@@ -357,6 +357,56 @@ module "stl" [system] {
...
+    export *
+    header "stdlib.h"
+  }
+  module "string_view" {
+    export *
+    header "string_view"
+  }
+  module "tgmath.h" {
```
Comment (Axel-Naumann) [inline build/unix/stl.modulemap]: This should probably go into `libc.modulemap`?

File: build/unix/stl.modulemap [hunk 3]
```diff
@@ -357,6 +357,56 @@ module "stl" [system] {
     export *
     header "vector"
   }
+
```
Comment (Axel-Naumann) [inline build/unix/stl.modulemap]: This empty line seems to be inconsistent with the rest of this file?

**Comment by yamaguchi1024** [comment]
Hi Axel,

We're going thought exercises for the Global Modules Index project, thanks for reviewing!

**Comment by arpi-r** [comment]
@oshadura, I have squashed the commits. 
All the headers used for generation of PCH are present except cassert. The stl.modulemap file says that it is not present to stay consistent with the OS X modulemap, so I did not add it.

**Comment by oshadura** [comment]
@phsft-bot build just on ROOT-ubuntu16/rtcxxmod with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by oshadura** [comment]
@arpi-r can you check please a failure, looks like <any> is not available as a part of std still.
`ksana@oksana-ThinkPad-E470:~/CERN_sources/root-modules/builds$ find /usr/ -name "any.h"
/usr/local/lib/python2.7/dist-packages/external/protobuf/src/google/protobuf/any.h
/usr/local/lib/python2.7/dist-packages/tensorflow/include/external/protobuf_archive/src/google/protobuf/any.h
/usr/local/lib/python2.7/dist-packages/tensorflow/include/google/protobuf/any.h
/usr/src/linux-headers-4.15.0-45-generic/include/config/wiznet/bus/any.h
/usr/src/linux-headers-4.15.0-46-generic/include/config/wiznet/bus/any.h
/usr/src/linux-headers-4.13.0-36-generic/include/config/wiznet/bus/any.h`

**Comment by arpi-r** [comment]
@oshadura I have included more headers than in the list of STL headers that are used for generation of PCH. Do I change stl.modulemap to contain headers only from that list? (This compiled in my system)

**Comment by oshadura** [comment]
@arpi-r, what I am asking is to fix a failure on our CI node ROOT-ubuntu16/rtcxxmod, where it was not able to find <any> header: `error: header 'any' not found`

**Comment by arpi-r** [comment]
@oshadura, I have made the changes and removed 'any' from stl.modulemap. Please check if it works now.

**Pull request merged**